BuildScript.Nuke/GenerateSbom/SolutionSbomGenerator.cs

```diff
@@ -0,0 +1,177 @@
+    _packageBlackList = packageBlackList;
+    _pathToPackageJson = pathToPackageJson;
+    _githubUsername = githubUsername;
+    _githubAccessToken = githubAccessToken;
+  }
+
+  public async Task Generate (Tool cycloneDxTool)
+  {
+    var uncleanedSbomFilename = "uncleaned-sbom.xml";
+
+    Directory.CreateDirectory(_workingDirectory);
+
+    cycloneDxTool.Invoke(
+        arguments:
+        $"-o . --disable-package-restore --exclude-dev --exclude-test-projects -gu {_githubUsername} -gt {_githubAccessToken} -f \"{uncleanedSbomFilename}\" {_solution.Path}",
+        workingDirectory: _workingDirectory
+    );
+
+    Log.Information("Cleaning Sbom...");
+
+    var cleanedSbomFileName = "cleaned-sbom.xml";
+    var cleaner = new Cleaner();
+
+    cleaner.CleanSolutionSbom(
+        _workingDirectory / uncleanedSbomFilename,
+        _solution.Path,
+        _packageBlackList,
+        _projectBlacklist,
+        _workingDirectory / cleanedSbomFileName);
+
+    Log.Information("Created cleaned Sbom '{_outputFile}'.", _workingDirectory / cleanedSbomFileName);
+
+    if (_pathToPackageJson == null)
+    {
+      Log.Information("No package json file specified, will therefore not create combined Sbom.");
+      return;
+    }
+
+    Log.Information("Downloading npm Sbom tool...");
+
+    var packageJsonFolder = _pathToPackageJson;
+    if (File.Exists(_pathToPackageJson))
+      packageJsonFolder  = Path.GetDirectoryName(_pathToPackageJson);
+
+    NpmTasks.Npm("install", packageJsonFolder);
+
+    NpmTasks.NpmInstall(_ => _
+        .SetPackages("@cyclonedx/cyclonedx-npm")
+        .SetProcessWorkingDirectory(_workingDirectory));
+
+    var npmSbomFileName = "npm-sbom.xml";
+
+    Log.Information("Creating npm Sbom...");
+
+    PowerShellTasks.PowerShell(_ => _
+        .SetCommand($".\\node_modules\\.bin\\cyclonedx-npm.ps1 {packageJsonFolder /"package.json"} --omit dev --output-format xml --output-file '{npmSbomFileName}'")
+        .SetProcessWorkingDirectory(_workingDirectory));
+
+    Log.Information("Created npm Sbom at '{npmSbomLocation}'.", _workingDirectory / npmSbomFileName);
+
+    var cycloneDXMergeToolPath = _workingDirectory / "cyclonedxtool" / "cyclonedx-cli.exe";
+    Directory.CreateDirectory(cycloneDXMergeToolPath.Parent);
+
+    if (!File.Exists(cycloneDXMergeToolPath))
+    {
+      Log.Information("CycloneDX merge tool not found in working directory. Will download tool from github.");
+      await DownloadCycloneDxMergeTool(cycloneDXMergeToolPath);
+    }
+
+    Log.Information("Merging npm and .net Sboms...");
+    PowerShellTasks.PowerShell(_ => _
+        .SetCommand($"{cycloneDXMergeToolPath} merge --input-files {cleanedSbomFileName} {npmSbomFileName} --output-file {_outputFile}")
+        .SetProcessWorkingDirectory(_workingDirectory));
+
+    Log.Information("Merged Sbom created at {0}", _outputFile);
+  }
+
+  private async Task DownloadCycloneDxMergeTool (AbsolutePath cycloneDXMergeToolPath)
```

Comment:
Question:
CycloneDX.Utils appears to also offer Merge capabilities. WOuld this be a better choice than using the console application?
https://github.com/CycloneDX/cyclonedx-dotnet-library/blob/main/tests/CycloneDX.Utils.Tests/MergeTests.cs